lib/ecto_enum.ex

```diff
@@ -143,4 +143,14 @@ defmodule EctoEnum do
     def dump(_), do: :error
   end
 
+  @spec validate_enum(Ecto.Changeset.t, atom, ((atom, String.t) -> String.t)) :: Ecto.Changeset.t
+  def validate_enum(changeset, field, error_formater \\ fn(field, value) -> "Value #{value} is not member of #{field} enum" end) do
+    type = changeset.types[field]
+    value = Ecto.Changeset.get_field(changeset,field)
+
+    case type.valid_value?(value) do
+      true -> changeset
+      _ -> Ecto.Changeset.add_error(changeset, field, error_formater.(field, value))
```

Comment:
@gjaldon related to something I fixed on Ecto (https://github.com/elixir-ecto/ecto/pull/2180), could you add  `validation: :enum` + whatever that formatter returns?

To be honest, I would prefer to do what Ecto validations do but it doesn't matter as long as you include the `validation` key so people could use it in the future.

https://github.com/elixir-ecto/ecto/blob/cff966d56f33d6b8ac8920913ee191b9c91e70f4/lib/ecto/changeset.ex#L1808

```elixir
defp message(opts, key \\ :message, default) do
    Keyword.get(opts, key, default)
  end

...

Changeset.add_error(changeset, field_name, message, [validation: :enum])
```